netCDF4/_netCDF4.pyx

```diff
@@ -2273,14 +2302,39 @@ version 4.1.2 or higher of the netcdf C lib, and rebuild netcdf4-python."""
         # view.obj is checked, ref on obj is decremented and obj will be null'd out
         PyBuffer_Release(&self._buffer)
 
+    IF HAS_NC_CREATE_MEM:
+        def _close_mem(self, check_err):
+            cdef int ierr
+            cdef NC_memio memio
+            ierr = nc_close_memio(self._grpid, &memio)
+
+            if check_err:
+                _ensure_nc_success(ierr)
+
+            self._isopen = 0 # indicates file already closed, checked by __dealloc__
+
+            # Only release buffer if close succeeded
+            # per impl of PyBuffer_Release: https://github.com/python/cpython/blob/master/Objects/abstract.c#L667
+            # view.obj is checked, ref on obj is decremented and obj will be null'd out
+            PyBuffer_Release(&self._buffer)
+
+            # return bytes representing in-memory dataset
+            return PyBytes_FromStringAndSize(<char *>memio.memory, memio.size)
```

Comment:
I think you may have a memory leak here (unless Cython manages this automatically somehow):
"WARNING, the returned memory is owned by the caller and so the caller is responsible for calling free() on that returned memory."
https://www.unidata.ucar.edu/software/netcdf/docs/md__Users_wfisher_Desktop_v4_86_82_netcdf-c_docs_inmemory.html

Reply:
you're right - that should be fixed now.